include/flexflow/parallel_ops/fused_parallel_op_params.h

```diff
@@ -0,0 +1,21 @@
+#ifndef _FLEXFLOW_FUSED_PARALLEL_OP_PARAMS_H
+#define _FLEXFLOW_COMBINE_PARAMS_OP_PARAMS_H
+
+namespace FlexFlow {
+
+struct FusedParallelOpParams {
+  std::vector<ParallelOpInfo> parallel_ops;
+  bool is_valid(ParallelTensorShape const &) const;
+};
+bool operator==(FusedParallelOpParams const &, FusedParallelOpParams const &);
+
+} // namespace FlexFlow
+
+namespace std {
+template <>
+struct hash<FlexFlow::FusedParallelOpParams> {
+  size_t operator()(FlexFlow::FusedParallelOpParams const &) const;
+}
```

Comment:
Why is there a semicolon after the struct specialization in other files but not here?

Reply:
That's odd. I would've thought that gets caught by the formatter. Just added it to the other parallel ops too.

Reply:
Is the semicolon required to compile? The formatter's role (at least according to our formatter) generally consists of taking syntactically correct code and producing correctly formatted syntactically correct code, so if the input code is broken it's kinda allowed to do whatever I think, which might(?) explain it? 

As a side note currently there is no validation that the code still builds after a commit, so if you can always validate locally that everything still builds before you merge.